graphql_compiler/compiler/expressions.py

```diff
@@ -641,8 +642,20 @@ def to_gremlin(self):
                              u'in Gremlin, so this is a bug. This function should not be called.')
 
     def to_cypher(self):
-        """Not implemented yet."""
-        raise NotImplementedError()
+        """Return a unicode object with the Cypher representation of this expression."""
+        self.validate()
+
+        _, field_name = self.fold_scope_location.get_location_name()
+        mark_name = emit_cypher.collected_vertex_list_name(
```

Comment:
`expressions.py` is calling into `emit_cypher` in two places here. On one hand, it's doing so from the `to_cypher()` method so maybe it's okay. On the other, it still feels kind of icky. Is there a place to put these two functions where we won't have to do this?

Reply:
I originally had the functions in `emit_cypher` since that has us write out `emit_cypher.function_name()`, so it's less likely someone goes and uses it for non-Cypher things in the future.

OTOH functions like `fold_scope_location_full_path_name` aren't necessarily Cypher-specific-- maybe at some later point it would be useful for new backends as well. Open to moving any non-Cypher-specific functions to `helpers.py`.

That doesn't solve the question of what to do with `collected_vertex_list_name` because that _is_ Cypher-specific-- in that case I'm not sure it would make sense in `helpers.py` because it's specific to one language. In that case we could probably keep `collected_vertex_list_name` where it is.

Reply:
follow-up point about `fold_scope_location_full_path_name`: PR #464 also uses this function in ir lowering so I definitely can see it going somewhere other than `emit_cypher` bc now it's crossing from emit to IR lowering and that shouldn't happen